plugins/xlsx-worbook-sink/src/plugin.ts

```diff
@@ -0,0 +1,175 @@
+
+    try {
+      await api.jobs.ack(jobId, {
+        info: "Starting job to write to Excel file",
+        progress: 10,
+      });
+
+      for (const sheet of sheets) {
+        try {
+          const { data } = await api.records.get(sheet.id);
+
+          const rows = R.pipe(
+            data.records,
+            R.map(({ values: row }) => {
+              return R.pipe(
+                Object.keys(row),
+                R.reduce((acc, colName) => {
+                  return {
+                    ...acc,
+                    [colName]: row[colName].value,
+                  };
+                }, {})
+              );
+            })
+          );
+
+          const worksheet = XLSX.utils.json_to_sheet(rows);
+
+          XLSX.utils.book_append_sheet(
+            workbook,
+            worksheet,
+            // Limit sheet name to 31 characters
+            sheet.name.substring(0, 31)
+          );
+        } catch (_getRecordsError: unknown) {
+          logError("Failed to fetch records for sheet with id: " + sheet.id);
+
+          return;
+        }
+      }
+    } catch (_jobAckError: unknown) {
+      logError("Failed to acknowledge job with id: " + jobId);
+
+      return;
+    }
+
+    const fileName = `Workbook-${currentEpoch()}.xlsx`;
+
+    try {
+      XLSX.writeFileXLSX(workbook, fileName);
+
+      if (opts.debug) {
+        logInfo("File written to disk");
+      }
+    } catch (_writeError: unknown) {
+      logError("Failed to write file to disk");
+
+      await api.jobs.fail(jobId, {
+        outcome: {
+          message:
+            "Job failed because it could not write the Excel Workbook to disk.",
+        },
+      });
+
+      return;
+    }
+
+    try {
+      const reader = fs.createReadStream(fileName);
+
+      await api.files.upload(reader, {
+        spaceId,
+        environmentId,
+        mode: "export",
+      });
+
+      reader.close();
+
+      if (opts.debug) {
+        logInfo(
+          `Excel document uploaded. View file at https://spaces.flatfile.com/space/${spaceId}/files?mode=export`
+        );
+      }
+    } catch (_uploadError: unknown) {
+      logError("Failed to upload file");
+
+      await api.jobs.fail(jobId, {
+        outcome: {
+          message: "Job failed because it could not upload Excel file.",
+        },
+      });
+
+      return;
+    }
+
+    try {
+      await api.jobs.complete(jobId, {
+        outcome: {
+          message:
+            'Data was successfully written to Excel file and uploaded. You can access the workbook in the "Available Downloads" section of the Files page in Flatfile.',
+        },
+      });
+
+      if (opts.debug) {
+        logInfo("Done");
+      }
+    } catch (_jobError: unknown) {
+      logError("Failed to complete job");
+
+      return;
+    }
+  } catch (_fetchSheetsError: unknown) {
+    logError("Failed to fetch sheets for workbook id: " + workbookId);
+
+    return;
+  }
+};
+
+const currentEpoch = (): string => {
```

Comment:
If we don't like epoch as a unique identifier on files, we can change it something else.